### PR TITLE
Add :xcodeproj param to set_build_number_repository action

### PR DESCRIPTION
**Pull request by AndrewSB**
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:

- [ ] Run `bundle exec rspec` from the subdirectory of each tool you modified. Alternatively, run `rake test_all` from the root directory.
- [x] Run `bundle exec rubocop -a` to ensure the code style is valid
- [x] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:

### CODE REVIEW

File: fastlane/lib/fastlane/actions/set_build_number_repository.rb [hunk 1]
```diff
@@ -13,7 +13,10 @@ def self.run(params)
           use_hg_revision_number: params[:use_hg_revision_number]
         )
 
-        Fastlane::Actions::IncrementBuildNumberAction.run(build_number: build_number)
+        Fastlane::Actions::IncrementBuildNumberAction.run(
+          build_number: build_number,
+          xcodeproj: params[:xcodeproj]
```
Comment (AndrewSB) [inline fastlane/lib/fastlane/actions/set_build_number_repository.rb]: not that familiar with ruby, not sure if I should be doing an if check to make sure `params[:xcodeproj]` isn't `nil` before passing it in.

File: fastlane/lib/fastlane/actions/set_build_number_repository.rb [hunk 2]
```diff
@@ -39,17 +42,24 @@ def self.available_options
                                        description: "Use hg revision number instead of hash (ignored for non-hg repos)",
                                        optional: true,
                                        is_string: false,
-                                       default_value: false)
+                                       default_value: false),
+          FastlaneCore::ConfigItem.new(key: :xcodeproj,
+                                       env_name: "XCODEPROJ_PATH",
+                                       description: "explicitly specify which xcodeproj to use",
+                                       optional: true)
```
Comment (mfurtak) [inline fastlane/lib/fastlane/actions/set_build_number_repository.rb]: The following would be good to add to the `ConfigItem` for some value validation:

```ruby
verify_block: proc do |value|
  path = File.expand_path(value)
  UI.user_error!("Please pass the path to the project, not the workspace") if path.end_with?(".xcworkspace")
  UI.user_error!("Could not find Xcode project at #{path}") unless Helper.is_test? || File.exist?(path)
end
```

File: fastlane/lib/fastlane/actions/set_build_number_repository.rb [hunk 3]
```diff
@@ -39,17 +42,24 @@ def self.available_options
...
-        ["pbrooks", "armadsen"]
+        ["pbrooks", "armadsen", "AndrewSB"]
       end
 
       def self.example_code
         [
-          'set_build_number_repository'
+          'set_build_number_repository',
+          'set_build_number_repository(
+            xcodeproj: "./path/to/MyApp.xcodeproj" # optional, by default it will set to every Xcodeproj found in the directory
```
Comment (mfurtak) [inline fastlane/lib/fastlane/actions/set_build_number_repository.rb]: Is that true? _Every_ .xcodeproj found?

File: fastlane/lib/fastlane/actions/set_build_number_repository.rb (hunk 1 above)
Comment (mfurtak) [inline fastlane/lib/fastlane/actions/set_build_number_repository.rb]: This should be OK 👍  `xcodeproj` is an optional param to the other action, so it handles the possibility that the value is `nil`

**Comment by mfurtak** [comment]
It would be good to get the `set_build_number_repository_spec.rb` updated to test the new possibility. Would you be interested in giving that a try?

File: fastlane/lib/fastlane/actions/set_build_number_repository.rb (hunk 3 above)
Comment (AndrewSB) [inline fastlane/lib/fastlane/actions/set_build_number_repository.rb]: It's definitely not 😂 thank you for catching that

File: fastlane/spec/actions_specs/set_build_number_repository_spec.rb [hunk 4]
```diff
@@ -22,6 +22,16 @@
 
         expect(Fastlane::Actions.lane_context[Fastlane::Actions::SharedValues::BUILD_NUMBER]).to match(/cd .* && agvtool new-version -all asd123/)
       end
+      
+      it "set build number with explicit xcodeproj" do 
+        result = Fastlane::FastFile.new.parse("lane :test do
+            set_build_number_repository(
+              xcodeproj: asd123/project.xcodeproj
```
Comment (AndrewSB) [inline fastlane/spec/actions_specs/set_build_number_repository_spec.rb]: @mfurtak I'm not sure what the path of the xcodeproj actually is, but other than that I think this test is good

**Comment by AndrewSB** [comment]
Also, I'm not sure whats causing CI to fail, it doesn't look as if I've touched the newly failing tests

**Comment by milch** [comment]
@AndrewSB We try to avoid this as much as possible, but sometimes failing tests make it into master (e.g. after merging two PRs without rebasing that break each other's tests, but not their own ones). 

Could you try rebasing your branch to the most current version of master to see if that fixes anything?

**Comment by mfurtak** [comment]
Continued in #7860 to rebase on master and fix the test.

**Pull request closed without merging**